pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.security;
+
+import java.util.List;
+
+import net.sourceforge.pmd.lang.apex.ast.ASTMethodCallExpression;
+import net.sourceforge.pmd.lang.apex.ast.ASTUserClass;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+
+/**
+ * Flags dangerous method calls, e.g. FinancialForce
+ * Configuration.disableTriggerCRUDSecurity()
+ * 
+ * 
+ * @author sergey.gorbaty
+ *
+ */
+public class ApexDangerousMethodsRule extends AbstractApexRule {
+	private static final String DISABLE_CRUD = "disableTriggerCRUDSecurity";
+	private static final String CONFIGURATION = "Configuration";
+
+	public ApexDangerousMethodsRule() {
+		setProperty(CODECLIMATE_CATEGORIES, new String[] { "Security" });
+		setProperty(CODECLIMATE_REMEDIATION_MULTIPLIER, 100);
+		setProperty(CODECLIMATE_BLOCK_HIGHLIGHTING, false);
+
```

Comment:
if you are only interested in visiting `ASTUserClass`, you could add here a call to `super.addRuleChainVisit(ASTUserClass.class);`.

This will allow the rule to be part of a rule chain, minimizing overhead of visiting the Abstract Syntax Tree by allowing a single transversal of the tree to be shared by multiple rules; and therefore making the application of this rule faster.

I am realizing this may apply to other rules in the security ruleset. Haven't thought of it earlier.